### PR TITLE
Enhance sscanf for regular expressions

### PR DESCRIPTION
**Pull request by Shyamakshi**
Current sscanf did not support regular expression parsing which is required for libcurl.

### CODE REVIEW

**Comment by Shyamakshi** [comment]
@juitem 
@philip8282

**Comment by seinfra** [comment]
Code Rule Check Result:
lib/libc/stdio/lib_sscanf.c:228: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:228: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:229: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:229: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:230: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:230: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:265: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:265: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:266: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:266: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:267: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:267: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:268: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:268: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:269: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:269: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:270: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:270: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:271: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:271: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:274: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:274: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:278: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:278: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:279: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:279: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:281: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:281: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:282: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:282: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:283: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:283: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:285: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:285: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:287: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:287: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:289: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:289: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:290: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:290: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:292: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:292: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:293: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:293: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
lib/libc/stdio/lib_sscanf.c:294: ERROR: [IDT_M_TAB] code indent should use tabs where possible
lib/libc/stdio/lib_sscanf.c:294: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@Shyamakshi Thank you for contribution. But please find our [commit rule](https://github.com/Samsung/TizenRT/wiki/Documentations#commit-rules).
Here are 3 commits, but two are duplicated, last is redundant.

File: lib/libc/stdio/lib_sscanf.c [hunk 1]
```diff
@@ -261,11 +261,40 @@ int vsscanf(FAR const char *buf, FAR const char *fmt, va_list ap)
...
+					const char *end;
+					if (width <= 0)
+						width = INT_MAX;
+
+					if (*++fmt == '^') {
+						complementary = true;
+						++fmt;
+					}
+					end = strchr((*fmt == ']') ? fmt + 1 : fmt, ']');
+					if (!end)
```
Comment (sunghan-chang) [inline lib/libc/stdio/lib_sscanf.c]: Please use brackets even though it has single statement

**Comment by Shyamakshi** [comment]
Will trigger another pull request due to duplicate commits.

**Pull request closed without merging**